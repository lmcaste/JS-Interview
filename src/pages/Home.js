import React, { Component } from 'react';
import {Link} from 'mobx-router';
import views from '../config/views';
import { inject, observer } from "mobx-react"
import Button from 'react-bootstrap/Button';

class Home extends Component {
  submitReview = (e) => {
    e.preventDefault();
		this.props.store.app.userRepositories(this.state.username);
    this.props.store.app.userOrganizations(this.state.username);
	};

  handleChange = event => {
    this.setState({ username: event.target.value });
  };

  render() { 
    const {store} = this.props;

    return (
      <div className="App">
        <h3 className="subtitle">Insert user name to get repositories and organizations in GitHub</h3>
        <div className="input-username">
        <input
          placeholder="Insert a username"
          type="text"
          name="username"
          label="username"
          value={store.username}
          onChange={this.handleChange} />
        <Link to={{
          pathname: "./list",
          data: [`${store.username}`]
        }} view={views.userlist} store={store}>
          <Button variant="raised" onClick={this.submitReview}> 
              Search
          </Button>
        </Link>
        </div>
      </div>
    );
  }
}
export default inject('store')(observer(Home));