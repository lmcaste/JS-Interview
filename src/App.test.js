import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Home from './pages/Home';

test('FocusInput matches snapshot', () => {
  const { container } = render(<Home store="store"/>)
  expect(container.firstChild).toMatchSnapshot();
});

test.skip('clicking on button increments counter', () => {
  const { getByText, getByTestId } = render(<Home store="store" />)
  fireEvent.click(getByText("Search"));
  expect(getByTestId("count").textContent).toBe("Clicked 2 times");
});

test.skip('calls "onChange" prop on button click', () => {
  const onChange = jest.fn();
  const { getByLabelText } = render(<Home store="store"/>);
  const upperInput = getByLabelText(username)
  fireEvent.change(upperInput, {target: {value: upper}})

  expect(onChange).toHaveBeenCalled();
});