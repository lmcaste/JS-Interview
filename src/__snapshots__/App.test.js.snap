// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FocusInput matches snapshot 1`] = `
<div
  class="App"
>
  <h3
    class="subtitle"
  >
    Insert user name to get repositories and organizations in GitHub
  </h3>
  <div
    class="input-username"
  >
    <input
      label="username"
      name="username"
      placeholder="Insert a username"
      type="text"
      value=""
    />
  </div>
</div>
`;
